simple-storage/src/main/java/com/sromku/simple/storage/ExternalStorage.java

```diff
@@ -81,9 +87,13 @@ protected String buildPath(String name) {
 	 * @return
 	 */
 	protected String buildPath(String directoryName, String fileName) {
-		String path = Environment.getExternalStorageDirectory().getAbsolutePath();
+		String path = buildAbsolutePath();
 		path = path + File.separator + directoryName + File.separator + fileName;
 		return path;
 	}
 
+	public ExternalStorage getPublicDirectory(String publicDirectory) {
```

Comment:
the method doesn't really return a public directory. No need for this method. It can be very confusing for consumer of this library. Or change it to private and to set<> because this is what it does. Something like:

```
private void setPublicDirectory(String publicDirectory) {
    mPublicDirectory = publicDirectory;
}
```